test/tests/p5.Envelope.js

```diff
@@ -0,0 +1,56 @@
+let expect = chai.expect;
+
+describe('p5.Envelop', function () {
+  let sf;
+
+  it('can be created and disposed', function () {
+    let envelope = new p5.Envelope();
+    envelope.dispose();
+  });
+
+  it('Testing setRange', function () {
+    let envelope = new p5.Envelope();
+    envelope.setRange(5, 6);
+    expect(envelope.aLevel).to.be.equal(5);
+    expect(envelope.rLevel).to.be.equal(6);
+    envelope.dispose();
+  });
+
+  it('Testing _setRampAD and setRampPercentages ', function () {
+    let envelope = new p5.Envelope();
+    envelope._setRampAD(5, 6);
+    envelope.setRampPercentages(5, 6);
+    expect(envelope._rampAttackTime).to.be.equal(5);
+    expect(envelope._rampDecayTime).to.be.equal(6);
+    expect(envelope._rampHighPercentage).to.be.equal(5);
+    expect(envelope._rampLowPercentage).to.be.equal(6);
+    expect(envelope._rampAttackTC).to.be.equal(0.27143405118953234);
+    expect(envelope._rampDecayTC).to.be.equal(600000000);
```

Comment:
similar to my comment above, since these are meant to be "private" properties that could change internally without the public-facing API of this class needing to change, it's best to find another way to test that the envelope is creating the correct shape.

Reply:
removed